apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -3827,6 +3832,212 @@ sc_ws_withdraw_and_close(Config, Origin) when Origin =:= initiator
+    Participants = proplists:get_value(participants, ConfigList),
+    Clients = proplists:get_value(channel_clients, ConfigList),
+    {SenderRole, AckRole} =
+        case Owner of
+            initiator -> {initiator, responder};
+            responder -> {responder, initiator}
+        end,
+    #{pub_key := SenderPubkey,
+      priv_key:= SenderPrivkey} = maps:get(SenderRole, Participants),
+    #{pub_key := AckPubkey,
+      priv_key:= AckPrivkey} = maps:get(AckRole, Participants),
+    SenderConnPid = maps:get(SenderRole, Clients),
+    AckConnPid = maps:get(AckRole, Clients),
+
+    ok = ?WS:register_test_for_channel_events(SenderConnPid, [sign, info, get]),
+    ok = ?WS:register_test_for_channel_events(AckConnPid, [sign, info, get]),
+
+    %% helper lambda for update
+    UpdateVolley =
+        fun() ->
+            UnsignedStateTx = channel_sign_tx(SenderConnPid, SenderPrivkey, <<"update">>),
+
+            % acknowledger signs update_ack
+            {ok, #{<<"event">> := <<"update">>}} = ?WS:wait_for_channel_event(AckConnPid, info),
+            UnsignedStateTx = channel_sign_tx(AckConnPid, AckPrivkey, <<"update_ack">>)
+        end,
+
+        
+
+    % trigger new contract
+    {UnsignedStateTx, Code} = create_contract_(TestName, SenderConnPid, UpdateVolley),
+
+    ContractPubKey = contract_id_from_create_update(SenderPubkey,
+                                                    UnsignedStateTx),
+
+    %% helper lambda for decoded result
+    GetDecodedResult =
+        fun(Type, UnsignedStateTx00) ->
+            {CB1, Tx1} = aetx:specialize_callback(UnsignedStateTx00),
+            CallRound = CB1:round(Tx1),
+            CallerAddress = aec_base58c:encode(account_pubkey, SenderPubkey),
+            ContractAddress = aec_base58c:encode(contract_pubkey, ContractPubKey),
+            GetCallResult =
+                fun(ConnPid) ->
+                    ?WS:send_tagged(ConnPid, <<"get">>, <<"contract_call">>,
+                                    #{contract   => ContractAddress,
+                                      caller     => CallerAddress,
+                                      round      => CallRound}),
+
+                    {ok, <<"contract_call">>, Res} = ?WS:wait_for_channel_event(ConnPid, get),
+                    Res
+                end,
+            CallRes = GetCallResult(SenderConnPid),
+            CallRes = GetCallResult(AckConnPid),
+            #{<<"caller_address">>    := CallerAddress,
+              <<"caller_nonce">>      := CallRound, 
+              <<"contract_address">>  := ContractAddress,
+              <<"gas_price">>         := _,
+              <<"gas_used">>          := _,
+              <<"height">>            := CallRound,
+              <<"return_type">>       := <<"ok">>,
+              <<"return_value">>      := ReturnValue} = CallRes,
+            {ok, 200, #{<<"data">> := #{<<"type">> := <<"word">>, <<"value">> := DecodedCallResult00}}} =
```

Comment:
This `<<"word">>` assumes which contracts this test helper function is being called with? Shall this be explicitly dependent on the test name instead?

Reply:
Addressed